server/gateway.go

```diff
@@ -2680,7 +2680,10 @@ func (s *Server) gatewayHandleSubjectNoInterest(c *client, acc *Account, accName
 	// If there is no subscription for this account, we would normally
 	// send an A-, however, if this account has the internal subscription
 	// for service reply, send a specific RS- for the subject instead.
+	// Need to grab the lock here since sublist can change during reload.
+	acc.mu.RLock()
 	hasSubs := acc.sl.Count() > 0
+	acc.mu.RUnlock()
```

Comment:
Feels like we can make is such that we only need to lock once instead of twice.

Reply:
ah yes... will update

Reply:
ok updated